fdbclient/IKnobCollection.cpp

```diff
@@ -95,6 +95,33 @@ IKnobCollection& IKnobCollection::getMutableGlobalKnobCollection() {
 	return *globalKnobCollection();
 }
 
+Optional<Error> IKnobCollection::setupKnobs(const std::vector<std::pair<std::string, std::string>>& knobs) {
```

Comment:
I am not certain about the use of `Optional` here. Why not directly throw the exception?

Reply:
Since all the caller would just check the error and throw.

Reply:
`fdbcli` doesn't throw, but I think I could change that to throw.